src/test/java/core/basesyntax/service/impl/DataProcessingServiceImplTest.java

```diff
@@ -0,0 +1,116 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.model.OperationType;
+import core.basesyntax.service.DataProcessingService;
+import core.basesyntax.service.FruitService;
+import core.basesyntax.storage.Storage;
+import core.basesyntax.strategy.OperationsHandler;
+import core.basesyntax.strategy.impl.BalanceOperationsHandler;
+import core.basesyntax.strategy.impl.PurchaseOperationsHandler;
+import core.basesyntax.strategy.impl.ReturnOperationsHandler;
+import core.basesyntax.strategy.impl.SupplyOperationsHandler;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class DataProcessingServiceImplTest {
+    private static DataProcessingService processingService;
+
+    @BeforeAll
+    static void beforeAll() {
+        Map<OperationType, OperationsHandler> operationStrategies = new HashMap<>();
+        operationStrategies.put(OperationType.BALANCE, new BalanceOperationsHandler());
+        operationStrategies.put(OperationType.SUPPLY, new SupplyOperationsHandler());
+        operationStrategies.put(OperationType.PURCHASE, new PurchaseOperationsHandler());
+        operationStrategies.put(OperationType.RETURN, new ReturnOperationsHandler());
+        FruitService fruitService = new FruitServiceImpl(operationStrategies);
+        processingService = new DataProcessingServiceImpl(fruitService, 0, 1, 2);
+    }
+
+    @AfterEach
+    void afterEach() {
+        Storage.STORAGE.clear();
+    }
+
+    @Test
+    void processing_successProcessedSupllyOperation_Ok() {

```

Comment:
in this case, you don't need to test in this style. I think that one test with different types of operations is enough for you and expect that no error occurs